File: src/main/java/kitchenpos/menus/tobe/domain/MenuName.java
package kitchenpos.menus.tobe.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;

@Embeddable
public class MenuName {

    @Column(name = "name", nullable = false)
    private String name;

    protected MenuName() {
    }

    public MenuName(final String name, final ProfanityChecker profanityChecker) {
        if (name == null || name.isEmpty()) {
            throw new IllegalArgumentException("메뉴명은 필수로 입력해야 합니다.");
        }
        if (profanityChecker.containsProfanity(name)) {
            throw new IllegalArgumentException("메뉴명에 욕설이 포함되어 있습니다.");
        }
        this.name = name;
    }

    public String value() {
        return name;
    }
}

File: src/main/java/kitchenpos/menus/tobe/domain/MenuName.kt
package kitchenpos.menus.tobe.domain

@JvmInline
value class MenuName private constructor(val value: String) {
    companion object {
        fun of(name: String, menuNameValidator: MenuNameValidator): MenuName {
            menuNameValidator.validate(name)

            return MenuName(name)
        }
    }
}
